src/modm/platform/gpio/stm32/module.lb

```diff
@@ -199,6 +216,17 @@ def validate(env):
                 raise ValidateException("Pin '{}' already in EXTI map!".format(str(num)))
             extimap[str(num)] = vec
 
+    # Compute the set of remapped pins
+    positions = defaultdict(list)
+    for p in driver["package"][0]["pin"]:
+        if p.get("type", "io") == "io":
+            name = p["name"][1:4].strip().lower()
+            positions[p["position"]].append( (name, p.get("variant", "") == "remap") )
+    # Only choose the GPIOs that are on the same package position
+    remapped_gpios = {pin[0]: pin[1]
+                          for pins in positions.values() if len(pins) > 1
```

Comment:
We only care about pin names that occupy the same position, which only have the `PA9 [PA11]` format. Hence `[1:4].strip()` is enough.

Reply:
Btw, the code used in modm-devices is this one:
```py
def pin_name(name):
    name = name[:4]
    if len(name) > 3 and not name[3].isdigit():
        name = name[:3]
    return (name[1:2].lower(), name[2:].lower())
```

Reply:
But I guess you're right, it would be more robust to use that code, incase of a future `PA9-BLUB [PA11]`.

Reply:
Your solution looks robust enough. It will deal well with `PB3__OSC32_IN`, `PB3OSC32IN` or similar naming atrocities … But never underestimate STs creativity. I am waiting for a French `PÉ1` next to `PE1`.